exonum-java-binding/core/rust/integration_tests/tests/runtime.rs

```diff
@@ -0,0 +1,410 @@
+    );
+
+    runtime.initialize(&blockchain);
+
+    let future = runtime.deploy_artifact(artifact_one.0.clone(), artifact_one.1.to_bytes());
+    let result = core.run(future);
+
+    assert!(result.is_ok(), "Deploy first artifact result is not Ok()");
+    assert!(runtime.is_artifact_deployed(&artifact_one.0));
+
+    let future = runtime.deploy_artifact(artifact_two.0.clone(), artifact_two.1.to_bytes());
+    let result = core.run(future);
+
+    assert!(result.is_ok(), "Deploy second artifact result is not Ok()");
+    assert!(runtime.is_artifact_deployed(&artifact_two.0));
+
+    runtime.shutdown();
+}
+
+#[test]
+fn initialize_one_artifact() {
+    let runtime = get_runtime(6305);
+
+    let service_instance_config = get_service_instance_config(
+        runtime.get_executor(),
+        1,
+        "artifact__test_name",
+        "artifact:test-name",
+        "1.0",
+        create_service_artifact_valid,
+    );
+    let instances = vec![service_instance_config.0];
+
+    let mut test_kit = TestKitBuilder::validator()
+        .with_additional_runtime(runtime)
+        .with_instances(instances)
+        .create();
+    test_kit.create_block();
+    test_kit.stop();
+}
+
+#[test]
+fn initialize_two_artifacts() {
+    let runtime = get_runtime(6306);
+
+    let service_instance_one_config = get_service_instance_config(
+        runtime.get_executor(),
+        1,
+        "artifact__test_name_one",
+        "artifact:test-name-one",
+        "1.0",
+        create_service_artifact_valid,
+    );
+    let service_instance_two_config = get_service_instance_config(
+        runtime.get_executor(),
+        2,
+        "artifact__test_name_two",
+        "artifact:test-name-two",
+        "1.0",
+        create_service_artifact_valid,
+    );
+    let instances = vec![service_instance_one_config.0, service_instance_two_config.0];
+
+    let mut test_kit = TestKitBuilder::validator()
+        .with_additional_runtime(runtime)
+        .with_instances(instances)
+        .create();
+    test_kit.create_block();
+    test_kit.stop();
+}
+
+#[test]
+fn initialize_non_instantiable_artifact() {
+    let runtime = get_runtime(6307);
+
+    let service_instance_config = get_service_instance_config(
+        runtime.get_executor(),
+        1,
+        "artifact__test_name",
+        "artifact:test-name",
+        "1.0",
+        create_service_artifact_non_instantiable_service,
+    );
+    let instances = vec![service_instance_config.0];
+
+    let test_kit_builder = TestKitBuilder::validator()
+        .with_additional_runtime(runtime)
+        .with_instances(instances);
+    assert_panics(
+        format!(
+            "{} {}",
+            "Could not find a suitable constructor in",
+            "com.exonum.binding.fakes.services.invalidservice.NonInstantiableService",
+        )
+        .as_str(),
+        || test_kit_builder.create(),
+    );
+}
+
+fn create_artifact<C>(
+    executor: &Executor,
+    artifact_name: &str,
```

Comment:
Wouldn't it be better to pass single `artifact_id` argument containing both `artifact_name` and `artifact_version`? In a form `artifact-name:1.0`

Reply:
See also a comment on what a name is (groupId:artifactId:version).